### PR TITLE
Use the default branch name configured on the repo

### PR DESCRIPTION
**Pull request by brucebolt**
We are currently hardcoding the default branch name `master` as a protected branch.

This is problematic in two respects:
 - The name of this branch name is racially insensitive (GDS are migrating towards `main`, in place of `master`
 - It stops us having a default branch with a different name

Instead we will now protect the branch that is configured by default on the repo in GitHub.

### CODE REVIEW

File: github/spec/features/configure_repos_spec.rb [hunk 1]
```diff
@@ -126,7 +127,7 @@ def given_theres_a_repo(archived: false,
             })
       .to_return(body: {}.to_json, status: archived ? 403 : 200)
 
-    @branch_protection_update = stub_request(:put, "https://api.github.com/repos/#{full_name}/branches/master/protection")
+    @branch_protection_update = stub_request(:put, "https://api.github.com/repos/#{full_name}/branches/main/protection")
```
Comment (barrucadu) [inline github/spec/features/configure_repos_spec.rb:130]: These tests should use `#{default_branch}`

File: github/spec/features/configure_repos_spec.rb [hunk 2]
```diff
@@ -209,7 +210,7 @@ def the_repo_has_ci_enabled(full_name: "alphagov/smart-sandwich", providers: ["j
       }
     }
 
-    expect(WebMock).to have_requested(:put, "https://api.github.com/repos/#{full_name}/branches/master/protection").
+    expect(WebMock).to have_requested(:put, "https://api.github.com/repos/#{full_name}/branches/main/protection").
```
Comment (barrucadu) [inline github/spec/features/configure_repos_spec.rb:213]: Here too

File: github/spec/features/configure_repos_spec.rb [hunk 3]
```diff
@@ -218,7 +219,7 @@ def the_repo_has_ci_disabled(full_name: "alphagov/smart-sandwich")
       required_status_checks: nil
     }
 
-    expect(WebMock).to have_requested(:put, "https://api.github.com/repos/#{full_name}/branches/master/protection").
+    expect(WebMock).to have_requested(:put, "https://api.github.com/repos/#{full_name}/branches/main/protection").
```
Comment (barrucadu) [inline github/spec/features/configure_repos_spec.rb:222]: And here

File: github/spec/features/configure_repos_spec.rb (hunk 2 above)
Comment (brucebolt) [reply github/spec/features/configure_repos_spec.rb]: Thanks, I didn't spot either of these

**Pull request merged**